src/main/java/qupath/ext/wsinfer/models/WSInferUtils.java

```diff
@@ -42,42 +40,23 @@ public class WSInferUtils {
 
     private static WSInferModelCollection cachedModelCollection;
 
-    static void downloadURLToFile(URL url, File file) {
+    static void downloadURLToFile(URL url, File file) throws IOException {
         ReadableByteChannel readableByteChannel = null;
```

Comment:
Does this channel need closed?
I wonder if there's a more modern way to get the file. I see there's a suggestion using `HTTPClient` [here](https://www.amitph.com/java-download-file-from-url/#using_java_httpclient), but even that seems a bit involved since [`BodySubscribers.ofFile`](https://docs.oracle.com/en/java/javase/20/docs/api/java.net.http/java/net/http/HttpResponse.BodySubscribers.html#ofFile(java.nio.file.Path)) exists (but I haven't actually tried it to see if it works).

Reply:
I assume there's some obvious and not boilerplate-ridden way to do something as trivial as download a file, but I didn't manage to find it

Reply:
This method was a bit of a mess, both channel and stream need closing I think.